pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -1106,6 +1130,15 @@ public LLRealtimeSegmentDataManager(RealtimeSegmentZKMetadata segmentZKMetadata,
 
     // Create record transformer
     _recordTransformer = CompositeTransformer.getDefaultTransformer(schema);
+
+    // Acquire semaphore to create Kafka consumers
+    try {
+      _partitionIdSemaphore.acquire();
+    } catch (InterruptedException e) {
+      String errorMsg = "InterruptedException when acquiring semaphore for Segment: " + _segmentNameStr;
+      segmentLogger.error(errorMsg);
```

Comment:
will this throw an exception and log an error when the service is being shutdown?

Reply:
It should be ok, since it just logs an error message before the process gets fully shutdown. Plus, this logic is in the constructor, the segment hasn't been fully instantiated yet.

Reply:
For now, this may be ok that we get an exception during shutdown (at times). Eventualy, we need a clean way of shutting down without throwing exceptions